java/ql/test/query-tests/security/CWE-552/UrlForwardTest.ql

```diff
@@ -0,0 +1,18 @@
+import java
+import TestUtilities.InlineExpectationsTest
+import semmle.code.java.security.UrlForwardQuery
+
+module UrlForwardTest implements TestSig {
+  string getARelevantTag() { result = "hasUrlForward" }
+
+  predicate hasActualResult(Location location, string element, string tag, string value) {
+    tag = "hasUrlForward" and
+    exists(UrlForwardFlow::PathNode sink | UrlForwardFlow::flowPath(_, sink) |
+      location = sink.getNode().getLocation() and
+      element = sink.getNode().toString() and
+      value = ""
+    )
+  }
+}
+
+import MakeTest<UrlForwardTest>
```

Comment:
If you want to simplify:

```suggestion
import TestUtilities.InlineFlowTest
import semmle.code.java.security.UrlForwardQuery
import TaintFlowTest<UrlForwardQuery >
```

(but then you'll need to change the expectation message to `$ hasTaintFlow` in the test cases)

Reply:
Updated in https://github.com/github/codeql/pull/14854/commits/2391fe7d89ff46cad715c09963c73ae6a8fd6f90.